lib/filewatch/watched_file.rb

```diff
@@ -0,0 +1,180 @@
+require "filewatch/buftok"
+
+module FileWatch
+  class WatchedFile
+    def self.new_initial(path, inode, stat = nil)
+      new(path, inode, stat, true)
+    end
+
+    def self.new_ongoing(path, inode, stat = nil)
+      new(path, inode, stat, false)
+    end
+
+    attr_reader :size, :inode, :state, :file, :buffer, :state_history
+    attr_reader :path, :filestat, :ignored_size, :accessed_at
+    attr_accessor :close_older, :ignore_older, :delimiter
+
+    def delimiter
+      @delimiter
+    end
+
+    def initialize(path, inode, stat, initial)
+      @path = path
+      @ignored_size = @size = 0
+      @inode = inode
+      @initial = initial
+      @state_history = []
+      @state = :watched
+      if (@filestat = stat).nil? && !@inode.nil?
+        restat
+      end
+      set_accessed_at
```

Comment:
this may serve a point to be here, but I'm thinking if it might be misleading in the scenario where neither @filestat nor inode are nil, you don't actually know when it was accessed.

on the other hand, is it worth maybe adding `set_accessed_at` as part of `restat`?